models/mli/model_gam.py

```diff
@@ -0,0 +1,242 @@
+    _testing_can_skip_failure = False  # ensure tested as if shouldn't fail
+
+    @staticmethod
+    def do_acceptance_test():
+        return True
+
+    def set_default_params(self, accuracy=None, time_tolerance=None,
+                           interpretability=None, **kwargs):
+        # Fill up parameters we care about
+        self.params = dict(random_state=kwargs.get("random_state", 1234),
+                           max_depth_duplication = None, n_estimators= 10,
+                           lam=0.1, max_iter=100)
+        
+    def mutate_params(self, accuracy=10, **kwargs):
+        
+        if accuracy > 8:
+            lam = [0, 0.001, 0.01, 0.1, 1.0, 3.0, 5.0, 10.0]
+            max_iter = [100, 1000]
+
+        elif accuracy >= 5:
+            lam = [0,  0.01, 0.1, 1.0, 10.0]
+            max_iter = [100]
+
+        else:
+            lam = [0,  0.01, 0.1, 1.0, 10.0]
+            max_iter = [100]
+
+        self.params["lam"] = np.random.choice(lam) 
+        self.params["max_iter"] = np.random.choice(max_iter) 
+
+    def _create_tmp_folder(self, logger):
+        # Create a temp folder to store files 
+        # Set the default value without context available (required to pass acceptance test)
+        tmp_folder = os.path.join(temporary_files_path, "%s_GAM_model_folder" % uuid.uuid4())
+        # Make a real tmp folder when experiment is available
+        if self.context and self.context.experiment_id:
+            tmp_folder = os.path.join(self.context.experiment_tmp_dir, "%s_GAM_model_folder" % uuid.uuid4())
+
+        # Now let's try to create that folder
+        try:
+            os.mkdir(tmp_folder)
+        except PermissionError:
+            # This not occur so log a warning
+            loggerwarning(logger, "GAM was denied temp folder creation rights")
+            tmp_folder = os.path.join(temporary_files_path, "%s_GAM_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+        except FileExistsError:
+            # We should never be here since temp dir name is expected to be unique
+            loggerwarning(logger, "GAM temp folder already exists")
+            tmp_folder = os.path.join(self.context.experiment_tmp_dir, "%s_GAM_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+        except:
+            # Revert to temporary file path
+            tmp_folder = os.path.join(temporary_files_path, "%s_GAM_model_folder" % uuid.uuid4())
+            os.mkdir(tmp_folder)
+
+        loggerinfo(logger, "GAM temp folder {}".format(tmp_folder))
+        return tmp_folder
+
+    def fit(self, X, y, sample_weight=None, eval_set=None, sample_weight_eval_set=None, **kwargs):
+        
+        orig_cols = list(X.names)
+       
+        import pandas as pd
+        import numpy as np
+        from sklearn.preprocessing import OneHotEncoder
+        from collections import Counter
+        from pygam import LinearGAM, LogisticGAM
```

Comment:
H2O-3 now has GAMs. Might be worth looking into.

Reply:
Looking into it...